File: app/src/main/java/com/llbt/meepwn/lincoinblock/main/model/TestItemModel.java
package com.llbt.meepwn.lincoinblock.main.model;

import android.databinding.ObservableField;

import com.llbt.meepwn.lincoinblock.framework.base.BaseModel;
import com.llbt.meepwn.lincoinblock.utils.network.json_parser.JsonParser;

/**
 * package: com.llbt.meepwn.lincoinblock.main.model
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/29 下午3:41
 * desc:
 */
public class TestItemModel extends BaseModel {

	private ObservableField<String> text = new ObservableField<>();

	public static TestItemModel initWithJsonString(String jsonString) {
		return JsonParser.parse(jsonString, TestItemModel.class);
	}

	public ObservableField<String> getText() {
		return text;
	}

	public void setText(ObservableField<String> text) {
		this.text = text;
	}

	@Override
	public String toString() {
		return "{" +
				"\"text\" : " + "\"" + text.get() + "\"" +
				"}";
	}
}

File: app/src/main/java/com/llbt/meepwn/lincoinblock/main/model/TestItemModel.kt
package com.llbt.meepwn.lincoinblock.main.model

import android.databinding.ObservableField
import com.llbt.meepwn.lincoinblock.framework.base.BaseModel
import com.llbt.meepwn.lincoinblock.utils.network.json_parser.JsonParser

/**
 * package: com.llbt.meepwn.lincoinblock
 * author: MeePwn
 * email: maybewaityou@gmail.com
 * github: https://github.com/maybewaityou
 * date: 16/4/5 下午5:39
 * desc:
 */
class TestItemModel : BaseModel() {

    var text = ObservableField<String>()

    companion object {
        fun initWithJsonString(jsonString: String?): TestItemModel? {
            return JsonParser.parse<TestItemModel>(jsonString, TestItemModel::class.java)
        }
    }

    override fun toString(): String {
        return "{\"text\" : \"${text.get()}\"}"
    }

}
